### PR TITLE
Pixel projectiles rebase.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
:cl:
tweak: Firearms now use pixel-precise projectile iteration, ported from Aurora.
/:cl:

Taking over #509 as @Andromeda-K22 is beset by the vagracies of life. 
**TODO:**
- [x] Fix linter complaints.
- [x] Fix ballistics muzzle flashes.
- [x] Clean up port.
- [x] Add lights to the full span of laser projectiles.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Would be good if people who aren't me could pull this down and have a play, let me know if anything else needs adjusting, but otherwise I think it's ready to go.

**Comment by comma** [comment]
ran around with it on local, nothing jumped out at me

**Comment by MistakeNot4892** [comment]
Mazian also said this is fine locally so 👍 ready to go.

**Comment by MistakeNot4892** [comment]
Bump.

File: code/__defines/flags.dm [hunk 1]
```diff
@@ -57,6 +57,7 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define PASS_FLAG_TABLE                   BITFLAG(0)
 #define PASS_FLAG_GLASS                   BITFLAG(1)
 #define PASS_FLAG_GRILLE                  BITFLAG(2)
+#define PASS_FLAG_MOB                     0x8
```
Comment (Lohikar) [inline code/__defines/flags.dm]: 👀

**Review by Lohikar — COMMENTED** [review]
Define needs updating, otherwise looks fine.

**Review by comma — DISMISSED** [review]

File: code/modules/projectiles/projectile/trace.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	var/obj/item/projectile/test/trace = new /obj/item/projectile/test(get_turf(firer)) //Making the test....
+
+	//Set the flags and pass flags to that of the real projectile...
+	trace.pass_flags = pass_flags
+
+	return trace.launch(target) //Test it!
+
+//"Tracing" projectile
+/obj/item/projectile/test //Used to see if you can hit them.
+	invisibility = 101 //Nope!  Can't see me!
```
Comment (Lohikar) [inline code/modules/projectiles/projectile/trace.dm]: `INVISIBILITY_ABSTRACT`

File: code/modules/projectiles/projectile/energy.dm [hunk 3]
```diff
@@ -15,7 +15,7 @@
 	damage = 5
 	agony = 20
 	life_span = 15 //if the shell hasn't hit anything after travelling this far it just explodes.
-	muzzle_type = /obj/effect/projectile/bullet/muzzle
+	muzzle_type = /obj/effect/projectile/muzzle/bullet
 	var/flash_range = 1
 	var/brightness = 7
 	var/light_colour = "#ffffff"
```
Comment (Lohikar) [inline code/modules/projectiles/projectile/energy.dm]: :pain:
(out of scope for this PR though)

File: code/modules/projectiles/projectile/beams.dm [hunk 4]
```diff
@@ -15,19 +15,31 @@
...
+/obj/item/projectile/beam/variable/after_move()
+	. = ..()
+	var/obj/effect/projectile/invislight/light = locate() in loc
+	if(light)
+		light.light_color = color
+		light.set_light(l_color = light.light_color)
+
+/obj/item/projectile/beam/variable/update_effect(var/obj/effect/projectile/effect)
+	effect.color = color
+	effect.light_color = color
```
Comment (Lohikar) [inline code/modules/projectiles/projectile/beams.dm]: This line seems kind of pointless with the below one.

**Review by Lohikar — APPROVED** [review]

**Pull request merged**